src/services/CompareUtils.jl

```diff
@@ -23,11 +24,28 @@ const GeneratedCompareUnion = Union{MeanMaxPPE, VariableNodeData, DFGNodeParams,
                               GenericFunctionNodeData,
                               DFGFactor, DFGFactorSummary, SkeletonDFGFactor}
 
-@generated function ==(x::T, y::T) where T <: GeneratedCompareUnion
-    ignored = [:_internalId]
-    mapreduce(n -> :(x.$n == y.$n), (a,b)->:($a && $b), setdiff(fieldnames(x), ignored))
+function _compVerbose(type, prop, x, y, ignored)
+  @debug "Comparing: $type.$prop: $x ?= $y"
+  # Propagate the ignores if it's a GeneratedCompareUnion
+  if typeof(x) <: GeneratedCompareUnion
+      !(==(x,y, ignored)) && @info("$type.$prop: $x != $y")
+      return ==(x,y, ignored)
+  else
+      # Otherwise ignore ignored
+      !(==(x,y)) && @info("$type.$prop: $x != $y")
+      return ==(x,y)
+  end
+end
+function ==(x::T, y::T, ignored::Vector{Symbol}=Symbol[]) where T <: GeneratedCompareUnion
+    mapreduce(
+      n -> (_compVerbose(T, n, getfield(x,n), getfield(y,n), ignored)),
+      (a,b)-> (a && b),
+      setdiff(fieldnames(typeof(x)), ignored))
 end
 
+# Based these on the calls below, need to be confirmed
+≈(x::V, y::V) where V <: DFGVariable = ==(x,y, [:attributes, :solverDataDict, :_internalId, :createdTimestamp, :lastUpdatedTimestamp])
+≈(x::F, y::F) where F <: DFGFactor = ==(x,y, [:fnc, :attributes, :_variableOrderSymbols, :_internalId])
```

Comment:
`≈` should call `≈` in the same way as `==`
`==` is the generic equality operator
`≈` `isapprox` is: Inexact equality comparison: true if `norm(x-y) <= max(atol, rtol*max(norm(x), norm(y)))`. 
It would be great if @dehann can write a `isapprox` for kde. We can dispatch on an `isparametric` trait to distinguish between them.

Reply:
JuliaRobotics/ApproxManifoldProducts.jl#31

Reply:
id suggest accepting the error at this point and will be filled in once 31 is closed.